lib/spack/docs/pipelines.rst

```diff
@@ -245,7 +280,21 @@ runners known to the gitlab instance.  For Docker executor type runners, the
 as well as an ``entrypoint`` to override whatever the default for that image is).
 For other types of runners the ``variables`` key will be useful to pass any
 information on to the runner that it needs to do its work (e.g. scheduler
-parameters, etc.).
+parameters, etc.).  Any ``variables`` provided here will be added, verbatim, to
+each job, unless the value takes on a special form.  If the value has the form
+``$env:<var_value>``, then that variable will be inserted with ``var_value``
+first sampled from the environment during job generation.  This can be useful
+for propagating generation-time variables which normally would not be available
+child build jobs.
+
+The ``runner-attributes`` section also allows users to supply custom ``script``,
+``before_script``, and ``after_script`` sections to be applied to every job
+scheduled on that runner.  This allows users to do any custom preparation or
+cleanup tasks that fit their particular workflow, as well as completely
+customize the rebuilding of a spec if they so choose.  We will never generate
+a ``before_script`` or ``after_script`` for jobs, but if you do not provide
+a custom ``script``, we will generate one for you that invokes
+``spack ci rebuild``.
```

Comment:
Two points:

 1. Consider rephrasing to remove the use of self-referential pronouns.  Consider passive voice.  (e.g.: "Neither before_script nor after_script sections are generated by default ...")

 1. You might not want to document the fact that there are no before or after scripts generated, lest they become part of the de-facto API surface.  Maybe just say that a user-provided section will never be overridden by anything we automatically generate.  This language gives us an out of we ever need to generate a default `before_script` or `after_script` section in the future.

Reply:
Is passive void desirable?

Reply:
Maybe instead of using the passive voice, I can replace `we` with `spack` in most instances?